script/make_utils/check_pytest_determinism.sh

```diff
@@ -70,4 +70,37 @@ echo ""
 echo "diff:"
 echo ""
 diff -u "${OUTPUT_DIRECTORY}/one.modified.txt" "${OUTPUT_DIRECTORY}/three.txt" --ignore-all-space --ignore-blank-lines --ignore-space-change
+
+# Run an execution and then check that the sub_seeds depends on the file name and the function name and parameters
+RANDOMLY_SEED=$RANDOMLY_SEED TEST=tests/seeding/test_seeding_system_file_a.py make pytest_one_single_cpu > "${OUTPUT_DIRECTORY}/seeds.txt"
+
+# This would not be readable
+# SC2181: Check exit code directly with e.g., 'if mycmd;', not indirectly with $?.
+# shellcheck disable=SC2181
+if [ $? -ne 0 ]
+then
+    echo "The commandline failed with:"
+    cat  "${OUTPUT_DIRECTORY}/seeds.txt"
+    exit 255
+fi
+
+RANDOMLY_SEED=$RANDOMLY_SEED TEST=tests/seeding/test_seeding_system_file_b.py make pytest_one_single_cpu >> "${OUTPUT_DIRECTORY}/seeds.txt"
+
+# This would not be readable
+# SC2181: Check exit code directly with e.g., 'if mycmd;', not indirectly with $?.
+# shellcheck disable=SC2181
+if [ $? -ne 0 ]
+then
+    echo "The commandline failed with:"
+    cat  "${OUTPUT_DIRECTORY}/seeds.txt"
+    exit 255
+fi
+
+
+grep "Output" "${OUTPUT_DIRECTORY}/seeds.txt" > "${OUTPUT_DIRECTORY}/seeds_outputs.txt"
+grep "sub_seed" "${OUTPUT_DIRECTORY}/seeds.txt" > "${OUTPUT_DIRECTORY}/seeds_sub_seed.txt"
+
+python script/make_utils/check_first_line_is_different_from_other_lines.py --file "${OUTPUT_DIRECTORY}/seeds_outputs.txt"
```

Comment:
could you just explain what you are checking here and why it's important that it passes ?

Reply:
also why just the first line and not the whole file ?

Reply:
adding a 

```
# In the following test, we:
#       log all Output's in seeds_outputs.txt, in a file of N lines: then, we'll check that
#           the first Output is different from the 2nd to N-th once
#       log all sub_seed's in seeds_sub_seed.txt, in a file of N lines: then, we'll check that
#           the first sub_seed is different from the 2nd to N-th once
#
# The goal of these test is to check that, really, seeds and outputs are different, ie that the
# filename or the function name or the parameters were really used in the seeding. Ideally, we 
# should check that all the outputs and seeds are different from each other, but it would be a long
# test (ie, being in N**2 [or ideally N log(N)] instead of N), so it's not worth it
```

It should answer all your questions, does it?

Reply:
ok great thanks !